### PR TITLE
added modal for confirmation

### PR DESCRIPTION
**Pull request by Devyani606**
added Modal instead of window.confirm()

### CODE REVIEW

File: src/modules/admin/defaultRules/DefaultRulesComponent/index.jsx [hunk 1]
```diff
@@ -139,5 +179,10 @@ DefaultRulesComponent.propTypes = {
   rules: PropTypes.string.isRequired,
   isRuleEdit: PropTypes.bool.isRequired,
   isLoading: PropTypes.bool.isRequired,
+  confirmationModal: PropTypes.bool.isRequired,
+  toggleConfirmation: PropTypes.func.isRequired,
+  targetId: PropTypes.string.isRequired,
```
Comment (onkar-josh) [inline src/modules/admin/defaultRules/DefaultRulesComponent/index.jsx:184]: is it of string type?

File: src/modules/admin/testCaseList/testCaseListComponent/index.jsx [hunk 2]
```diff
@@ -62,6 +102,10 @@ TestCaseListComponent.propTypes = {
   testCases: PropTypes.shape().isRequired,
   handleOnTestCaseEdit: PropTypes.func.isRequired,
   handleOnTestCaseDelete: PropTypes.func.isRequired,
+  confirmationModal: PropTypes.bool.isRequired,
+  toggleConfirmation: PropTypes.func.isRequired,
+  targetId: PropTypes.string.isRequired,
```
Comment (onkar-josh) [inline src/modules/admin/testCaseList/testCaseListComponent/index.jsx:107]: same as above id should be of number type

File: src/modules/admin/defaultRules/DefaultRulesComponent/index.jsx (hunk 1 above)
Comment (Devyani606) [reply src/modules/admin/defaultRules/DefaultRulesComponent/index.jsx:184]: number

File: src/modules/admin/testCaseList/testCaseListComponent/index.jsx (hunk 2 above)
Comment (Devyani606) [reply src/modules/admin/testCaseList/testCaseListComponent/index.jsx:107]: ok

**Pull request merged**